### DISCUSSION TITLE
Uable to get syntax highlight for mojo code, after add mojo in nvim_lspconfig

### DISCUSSION THREAD

**Discussion by hopezh** [LazyVim/LazyVim#1767] [Q&A]
As shown below, I added `mojo` in `nvim-lspconfig`, following the [example code of lazyvim](https://www.lazyvim.org/configuration/examples). 

However, I'm not getting syntax highlighting for `mojo` code, although I can execute it (both the latest versions of `modular` and `mojo` are installed). 

May I ask if I missed anything here to get **mojo lsp** to work? 

![2023-10-17-16-33-40](https://github.com/LazyVim/LazyVim/assets/23435810/c0a75841-e509-4a8b-aab0-511f51766905)

![2023-10-17-16-35-27](https://github.com/LazyVim/LazyVim/assets/23435810/ad616419-dff7-4b2d-af17-fd3a54fac769)

**Comment by salvaft** [comment]
Syntax highlighting has more to do with TreeSitter. Have you tried TSInstall?

**Reply by hopezh** [reply]
I tried `TSInstall mojo`, and there is an error message, `Installation not possible`. 

It seems that `mojo` is not on the `Supported Languages` list...

I wonder how to get lsp support for `mojo`in `NeoVim`.

**Comment by joelflaig** [comment]
For syntax highlighting you should try out https://github.com/ovikrai/mojo-syntax.
LSP support and the extension I mentioned both worked well for me, as shown below.
For LSP I used 
```lua
require 'lspconfig'.mojo.setup {}
```
in  the `~/.config/nvim/init.lua` file.

![image](https://github.com/LazyVim/LazyVim/assets/126613308/d6911feb-f959-4dac-b4d9-a2442f72ddde)
